baybe/simulation/lookup.py

```diff
@@ -1,201 +1,301 @@
+    queries: pd.DataFrame,
+    targets: Collection[Target],
+    lookup: Callable,
+) -> None:
+    """Look up target values by querying a callable."""
+    # TODO: Currently, the alignment of return values to targets is based on the
+    #   column ordering, which is not robust. Instead, the callable should return
+    #   a dataframe with properly labeled columns.
 
-    # Compute the target values via a callable
-    elif isinstance(lookup, Callable):
-        # TODO: Currently, the alignment of return values to targets is based on the
-        #   column ordering, which is not robust. Instead, the callable should return
-        #   a dataframe with properly labeled columns.
-
-        # Since the return of a lookup function is a tuple, the following code stores
-        # tuples of floats in a single column with label 0:
-        measured_targets = queries.apply(lambda x: lookup(*x.values), axis=1).to_frame()
-        # We transform this column to a DataFrame in which there is an individual
-        # column for each of the targets....
-        split_target_columns = pd.DataFrame(
-            measured_targets[0].to_list(), index=measured_targets.index
+    # Since the return of a lookup function is a tuple, the following code stores
+    # tuples of floats in a single column with label 0:
+    measured_targets = queries.apply(lambda x: lookup(*x.values), axis=1).to_frame()
+    # We transform this column to a DataFrame in which there is an individual
+    # column for each of the targets....
+    split_target_columns = pd.DataFrame(
+        measured_targets[0].to_list(), index=measured_targets.index
+    )
+    # ... and assign this to measured_targets in order to have one column per target
+    measured_targets[split_target_columns.columns] = split_target_columns
+    if measured_targets.shape[1] != len(targets):
+        raise AssertionError(
+            "If you use an analytical function as lookup, make sure "
+            "the configuration has the right amount of targets "
+            "specified."
         )
-        # ... and assign this to measured_targets in order to have one column per target
-        measured_targets[split_target_columns.columns] = split_target_columns
-        if measured_targets.shape[1] != len(campaign.targets):
-            raise AssertionError(
-                "If you use an analytical function as lookup, make sure "
-                "the configuration has the right amount of targets "
-                "specified."
+    for k_target, target in enumerate(targets):
+        queries[target.name] = measured_targets.iloc[:, k_target]
+
+
+def _look_up_targets_from_dataframe(
+    queries: pd.DataFrame,
+    targets: Collection[Target],
+    lookup: pd.DataFrame,
+    impute_mode: Literal[
+        "error", "worst", "best", "mean", "random", "ignore"
+    ] = "error",
+) -> None:
+    """Look up target values from a dataframe."""
+    # # IMPROVE: Although it's not too important for a simulation, this
```

Comment:
Why the double #?

Reply:
was there in the old code, not touching content in this PR